src/main/java/com/hubspot/jinjava/util/ObjectIterator.java

```diff
@@ -55,4 +50,16 @@ public static ForLoop getLoop(Object obj) {
     // others
     return new ForLoop(Iterators.singletonIterator(obj), 1);
   }
+
+  @SuppressWarnings("unchecked")
+  public static ForLoop getLoop(Object obj, boolean iterateOverMapKeys) {
```

Comment:
instead of passing this down through all these methods, could you just use the static `JinjavaInterpreter.getCurrent()` to get a hold of the interpreter and the config?